torch/_higher_order_ops/utils.py

```diff
@@ -378,3 +386,37 @@ def _stack_pytree(pytrees):
         else:
             raise RuntimeError(f"Cannot stack {leaves}.")
     return pytree.tree_unflatten(stacked_out, out_spec)
+
+
+# We cannot call save_for_backward for symints. This helper function
+# can be used to save symints as a direct attributes in ctx.
+def save_tensors_and_symints_for_backward(ctx, args):
+    assert all(isinstance(arg, (torch.Tensor, torch.SymInt, int)) for arg in args), args
+    partitioned_args: List[Any] = [[], []]
+    pos = []
+    for i, arg in enumerate(args):
+        idx = 0 if isinstance(arg, torch.Tensor) else 1
+        partitioned_args[idx].append(arg)
+        pos.append(idx)
+
+    assert getattr(ctx, "sym_int_args", None) is None, "ctx already has sym_int_args"
+    assert getattr(ctx, "pos", None) is None, "ctx already has pos"
+    ctx.save_for_backward(*partitioned_args[0])
+    ctx.sym_int_args = partitioned_args[1]
+    ctx.pos = pos
```

Comment:
Can you run through an example of how this works in a comment here? This has high mental overload right now lol